lib/logstop.rb

```diff
@@ -14,19 +14,34 @@ module Logstop
   SSN_REGEX = /\b\d{3}[\s+-]\d{2}[\s+-]\d{4}\b/
   URL_PASSWORD_REGEX = /((?:\/\/|%2F%2F)\S+(?::|%3A))\S+(@|%40)/
 
-  def self.scrub(msg, ip: false, scrubber: nil)
+  DEFAULT_OPTIONS = {
+    password: true,
+    email: true,
+    credit_card: true,
+    phone: true,
+    ssn: true,
+    ip: false,
+    scrubber: nil,
+  }.freeze
+
+  def self.scrub(msg, **options)
     msg = msg.to_s.dup
+    if unknown_option_key = options.keys.find { |option_key| !DEFAULT_OPTIONS.include?(option_key) }
+      raise ArgumentError, "unknown option #{unknown_option_key}"
+    end
+    options = DEFAULT_OPTIONS.merge(options)
 
     # order filters are applied is important
-    msg.gsub!(URL_PASSWORD_REGEX, FILTERED_URL_STR)
-    msg.gsub!(EMAIL_REGEX, FILTERED_STR)
-    msg.gsub!(CREDIT_CARD_REGEX, FILTERED_STR)
-    msg.gsub!(CREDIT_CARD_REGEX_DELIMITERS, FILTERED_STR)
-    msg.gsub!(PHONE_REGEX, FILTERED_STR)
-    msg.gsub!(SSN_REGEX, FILTERED_STR)
+    msg.gsub!(URL_PASSWORD_REGEX, FILTERED_URL_STR) if options[:password]
```

Comment:
Don't think there is a case to separate this out yet so the if's seem fine to me.

Reply:
I just wanted to make it as flexible as possible - I am looking from the perspective of not only addressing our current need, but to help the gem become more flexible overall